File: python/Data Structures and Algorithms in Python Book/oop/arithmeticprogression.py
from progression import Progression

class ArithmeticProgression(Progression):

    def __init__(self, start=0, increment=1):
        super().__init__(start)
        self._increment = increment

    def _advance(self):
       self._current += self._increment
       


if __name__ == "__main__":
    aritmetic_progression = ArithmeticProgression(start = 1, increment=3)
    aritmetic_progression.print_progression(20)

File: java/ood/ArithmeticProgression.java
package ood;

public class ArithmeticProgression extends Progression {
    // store step size
    protected long increment;

    /** Construct arithmetic progression 0, 1, 2, ... */
    public ArithmeticProgression() {
        this(0, 1);                                                     // start at 0 with increment of 1
    } 

    /** Construct arithmetic progression 0, stepsize, 2 * stepsize, ... */
    public ArithmeticProgression(long stepsize) {
        this(0, stepsize);                                              // start at 0
    } 

    /** Construct arithmetic progression with aribtrary start and increment */
    public ArithmeticProgression(long start, long stepsize) {
        super(start);
        increment = stepsize;
    }

    @Override
    protected void advance() {
        current += increment;
    }
}
